packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx

```diff
@@ -0,0 +1,151 @@
+import { FilterableItem } from '@lightdash/common';
+import {
+    Highlight,
+    Loader,
+    MultiSelect,
+    MultiSelectProps,
+    Text,
+} from '@mantine/core';
+import { FC, ReactNode, useCallback, useMemo, useState } from 'react';
+
+import {
+    MAX_AUTOCOMPLETE_RESULTS,
+    useFieldValues,
+} from '../../../../../hooks/useFieldValues';
+import { useFiltersContext } from '../../FiltersProvider';
+import { mergeUniqueValues } from './autoCompleteUtils';
+
+type Props = Pick<MultiSelectProps, 'disabled' | 'placeholder'> & {
+    filterId: string;
+    field: FilterableItem;
+    values: string[];
+    suggestions: string[];
+    onChange: (values: string[]) => void;
+};
+
+const FilterMultiAutoComplete: FC<Props> = ({
+    filterId,
+    values,
+    field,
+    suggestions: initialSuggestionData,
+    disabled,
+    onChange,
+    placeholder,
+}) => {
+    const { projectUuid, getAutocompleteFilterGroup } = useFiltersContext();
+    if (!projectUuid) {
+        throw new Error('projectUuid is required in FiltersProvider');
+    }
+
+    const [search, setSearch] = useState('');
+
+    const autocompleteFilterGroup = useMemo(
+        () => getAutocompleteFilterGroup(filterId, field),
+        [field, filterId, getAutocompleteFilterGroup],
+    );
+
+    const { isLoading, results: resultsSet } = useFieldValues(
+        search,
+        initialSuggestionData,
+        projectUuid,
+        field,
+        autocompleteFilterGroup,
+        true,
+        { refetchOnMount: 'always' },
+    );
+
+    const results = useMemo(() => [...resultsSet], [resultsSet]);
```

Comment:
No big deal, but does this useMemo do anything? It seems like it just keeps an updated copy of the results, but you could just use them directly, right? Maybe I am missing something and this is critical :D

Reply:
oh it returns a set not an array. I think it was like that before let me check

Reply:
No problem, it seems ok, just a little strange